titus-server-master/src/test/java/com/netflix/titus/master/jobmanager/service/integration/MoveTaskTest.java

```diff
@@ -88,10 +90,34 @@ public void testMoveWithInvalidTargetJob() {
         }
     }
 
+    @Test
+    public void testMoveWithIncompatibleTargetJob() {
+        JobDescriptor<ServiceJobExt> jobDescriptor = oneTaskServiceJobDescriptor();
+        JobScenarioBuilder<ServiceJobExt> sourceJobBuilder = startNewJob(jobDescriptor);
+        String sourceJobId = sourceJobBuilder.getJobId();
+        JobDescriptor<ServiceJobExt> incompatible = jobDescriptor.but(descriptor ->
+                descriptor.getContainer().but(container -> container.getImage().toBuilder()
+                        .withName("other/image")
+                        .build()
+                )
+        );
+        String targetJobId = startNewJob(incompatible).getJobId();
+
+        try {
+            sourceJobBuilder.advance()
+                    .moveTask(0, 0, sourceJobId, targetJobId)
+                    .expectJobEvent(job -> assertThat(JobFunctions.getJobDesiredSize(job)).isEqualTo(0));
```

Comment:
Is the last step needed? Exception should be thrown from the `moveTask` step?

Reply:
True, copy-and-pasta. I will fix.